src/Raven.Server/Documents/Handlers/Processors/Subscriptions/SubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,75 @@
+        ChangeVectorValidationResult changeVectorValidationResult)
+    {
+        return CreateSubscription(requestHandler, databaseName, id, disabled, options, changeVectorValidationResult);
+    }
+
+    protected override ValueTask GetResponsibleNodesAndWaitForExecutionAsync(SubscriptionsHandler requestHandler, string name, long index)
+    {
+        return GetResponsibleNodesAndWaitForExecution(requestHandler, name, index);
+    }
+
+    protected override void ThrowOnRevisionsIfNeeded(bool subRevisions)
+    {
+        // no-op
+    }
+
+    internal static Task<ChangeVectorValidationResult> TryValidateChangeVector(SubscriptionsHandler requestHandler, TransactionOperationContext context, SubscriptionCreationOptions options, SubscriptionConnection.ParsedSubscription sub)
+    {
+        if (Enum.TryParse(options.ChangeVector, out Constants.Documents.SubscriptionChangeVectorSpecialStates changeVectorSpecialValue))
+        {
+            switch (changeVectorSpecialValue)
+            {
+                case Constants.Documents.SubscriptionChangeVectorSpecialStates.BeginningOfTime:
+                    return Task.FromResult(new ChangeVectorValidationResult() { ChangeVectorsCollection = null, InitialChangeVector = null });
+                case Constants.Documents.SubscriptionChangeVectorSpecialStates.LastDocument:
+                    return Task.FromResult(new ChangeVectorValidationResult()
+                    {
+                        ChangeVectorsCollection = null,
+                        InitialChangeVector = requestHandler.Database.DocumentsStorage.GetLastDocumentChangeVector(context.Transaction.InnerTransaction, context, sub.Collection)
+                    });
+            }
+        }
+
+        return Task.FromResult(new ChangeVectorValidationResult() { ChangeVectorsCollection = null, InitialChangeVector = options.ChangeVector });
+    }
+
+    public static Task<(long Index, long SubscriptionId)> CreateSubscription(SubscriptionsHandler requestHandler, string databaseName, long? id, bool? disabled, SubscriptionCreationOptions options, ChangeVectorValidationResult changeVectorValidationResult)
+    {
+        options.ChangeVector = changeVectorValidationResult.InitialChangeVector;
+        return requestHandler.Database.SubscriptionStorage.PutSubscription(options, requestHandler.GetRaftRequestIdFromQuery(), id, disabled, options.MentorNode);
+    }
+
+    public static async ValueTask GetResponsibleNodesAndWaitForExecution(DatabaseRequestHandler requestHandler, string name, long index)
+    {
+        using (requestHandler.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            // need to wait on the relevant remote node
+            var node = requestHandler.Database.SubscriptionStorage.GetResponsibleNode(context, name);
+            await WaitForSubscriptionExecutionAsync(requestHandler, context, index, node);
```

Comment:
you are holding a read tx while waiting for subscription execution? can we move this out of the tx scope?

Reply:
done